test/integration/userManagementFilters.test.js

```diff
@@ -0,0 +1,315 @@
+const chai = require("chai");
+const { expect } = chai;
+const chaiHttp = require("chai-http");
+
+const app = require("../../server");
+const authService = require("../../services/authService");
+const addUser = require("../utils/addUser");
+const cleanDb = require("../utils/cleanDb");
+// Import fixtures
+const userData = require("../fixtures/user/user")();
+const { generateUserStatusData } = require("../fixtures/userStatus/userStatus");
+
+const config = require("config");
+const { updateUserStatus } = require("../../models/userStatus");
+const { addTag } = require("../../models/tags");
+const { addLevel } = require("../../models/levels");
+const { addTagsToItem } = require("../../models/items");
+
+const cookieName = config.get("userToken.cookieName");
+
+chai.use(chaiHttp);
+
+describe("UserManagement Filters", function () {
+  let jwt;
+  let userId = "";
+  let oooUser;
+  let idleUser;
+  let activeUser;
+  let tagIdFE;
+  let tagIdBE;
+  let levelId1;
+  let levelId2;
+
+  before(async function () {
+    const updatedAtDate = Date.now();
+    const untilDate = updatedAtDate + 16 * 24 * 60 * 60 * 1000;
+    userId = await addUser();
+    jwt = authService.generateAuthToken({ userId });
+    oooUser = await addUser(userData[0]);
+    await updateUserStatus(oooUser, generateUserStatusData("OOO", updatedAtDate, untilDate));
+    idleUser = await addUser(userData[1]);
+    await updateUserStatus(idleUser, generateUserStatusData("IDLE", updatedAtDate, updatedAtDate));
+    activeUser = await addUser(userData[8]);
+    await updateUserStatus(activeUser, generateUserStatusData("ACTIVE", updatedAtDate, updatedAtDate));
+
+    // creating tag and levels
+    const { id: id1 } = await addTag({
+      name: "Vue JS",
+      type: "SKILL",
+      reason: "Web Dev FE",
+    });
+    tagIdFE = id1;
+    const { id2 } = await addTag({
+      name: "Node JS",
+      type: "SKILL",
+      reason: "Web Dev BE",
+    });
+    tagIdBE = id2;
+    const { id3 } = await addLevel({
+      name: "level 1",
+      value: 1,
+    });
+    levelId1 = id3;
+    const { id4 } = await addLevel({
+      name: "level 2",
+      value: 2,
+    });
+    levelId2 = id4;
+
+    // creating items
+    addTagsToItem({
```

Comment:
You can mock/fake the `addTagsToItem` model function and make it resolve this set of items, Instead of actually calling the model function.

Reply:
Similar to `addTagsToItem`, you can also `addTag` and `addLevel` and resolve set of items.

Reply:
While we can certainly mock/fake the `addTagsToItem` and other model function to return a set of response, but this approach may not provide us with the expected response for the assertions. Please correct me if I am incorrect.

Reply:
I think the opposite, as it won't make a unexpected response for assertion.